internal/injector/check.go

```diff
@@ -29,7 +32,18 @@ func (i *Injector) typeCheck(fset *token.FileSet, files []*ast.File) (types.Info
 	}
 	checker := types.NewChecker(&checkerCfg, fset, pkg, &typeInfo)
 
-	if err := checker.Files(files); err != nil {
+	astFiles := make([]*ast.File, len(files))
+	for i, file := range files {
+		astFiles[i] = file.AstFile
+	}
+
+	if err := checker.Files(astFiles); err != nil {
+		// This is a workaround for the fact that the Go type checker does not return a specific unexported error type
+		// TODO: Ask better error typing from the Go team for the go/types package
+		if strings.Contains("package requires newer Go version", err.Error()) {
+			return types.Info{}, fmt.Errorf("orchestrion was built with Go version %s but package %q requires a newer go version, please reinstall and pin orchestrion with a newer Go version: type-checking files: %w", runtime.Version(), i.ImportPath, err)
+		}
```

Comment:
Should have a test exercising this code path... otherwise it's just good intentions.

Reply:
Please advise me how you would test this good sir

Reply:
Probably just create a fake package that declares `go 1.{runtime.VERSION + 1}`, and try to build it w/ `GOTOOLCHAIN=local`, etc...?